### PR TITLE
hippy-vue, hippy-react support HMR

### PR DESCRIPTION
**Pull request by shenchaoran**
Before submitting a new pull request, please make sure:

- [x] Test cases have been added/updated/passed for the code you will submit.
- [x] Documentation has added or updated.
- [x] Commit message is following the [Convention Commit](https://conventionalcommits.org/) guideline with maximum 72 characters.
- [x] Squash the repeat code commits, short patches are welcome.

### CODE REVIEW

File: packages/hippy-vue/src/runtime/websocket.js [hunk 1]
```diff
@@ -86,6 +81,7 @@ class WebSocket {
       url,
     };
     Native.callNativeWithPromise(WEB_SOCKET_MODULE_NAME, 'connect', params).then((resp) => {
+      console.log(`connect to ws url: ${url}`, resp);
```
Comment (zoomchan-cxj) [inline packages/hippy-vue/src/runtime/websocket.js:84]: no console

File: examples/hippy-react-demo/scripts/hippy-webpack.dev.js [hunk 2]
```diff
@@ -2,20 +2,34 @@ const fs = require('fs');
...
+      overlay: false,
+    },
+  },
   entry: {
     index: ['regenerator-runtime', path.resolve(pkg.main), '@hippy/hippy-live-reload-polyfill'],
   },
   output: {
     filename: 'index.bundle',
     strictModuleExceptionHandling: true,
+    publicPath: 'http://localhost:38989/',
```
Comment (zoomchan-cxj) [inline examples/hippy-react-demo/scripts/hippy-webpack.dev.js:32]: why use publicPath

Comment (shenchaoran) [reply examples/hippy-react-demo/scripts/hippy-webpack.dev.js]: had remove this config, add `hotManifestPublicPath` option to `HippyHMRPlugin`

File: packages/hippy-vue/src/runtime/websocket.js (hunk 1 above)
Comment (shenchaoran) [reply packages/hippy-vue/src/runtime/websocket.js]: removed

File: examples/hippy-react-demo/package.json [hunk 3]
```diff
@@ -8,23 +8,23 @@
...
-    "hippy:dev": "webpack --config ./scripts/hippy-webpack.dev.js",
-    "hippy:vendor": "webpack --config ./scripts/hippy-webpack.ios-vendor.js --config ./scripts/hippy-webpack.android-vendor.js",
-    "hippy:build": "webpack --config ./scripts/hippy-webpack.ios.js --config ./scripts/hippy-webpack.android.js"
+    "serve": "cross-env NODE_ENV=development webpack-dev-server --mode=development --config ./scripts/hippy-webpack.web.js",
+    "build": "cross-env NODE_ENV=production webpack --mode=production --config ./scripts/hippy-webpack.web.js",
+    "hippy:local-debug": "cross-env NODE_ENV=development node ../../packages/hippy-debug-server/index.js -c ./scripts/hippy-webpack.dev.js",
+    "hippy:debug": "cross-env NODE_ENV=development hippy-debug -c ./scripts/hippy-webpack.dev.js",
+    "hippy:dev": "cross-env NODE_ENV=development webpack --config ./scripts/hippy-webpack.dev.js",
+    "hippy:vendor": "cross-env NODE_ENV=production webpack --config ./scripts/hippy-webpack.ios-vendor.js --config ./scripts/hippy-webpack.android-vendor.js",
+    "hippy:build": "cross-env NODE_ENV=production webpack --config ./scripts/hippy-webpack.ios.js --config ./scripts/hippy-webpack.android.js"
```
Comment (zoomchan-cxj) [inline examples/hippy-react-demo/package.json:17]: cross-env NODE_ENV=production 移除

File: examples/hippy-react-demo/package.json [hunk 4]
```diff
@@ -46,13 +46,17 @@
     "@babel/preset-react": "^7.9.4",
     "@hippy/debug-server": "^2.1.2",
     "@hippy/hippy-dynamic-import-plugin": "^2.0.0",
+    "@hippy/hippy-hmr-plugin": "^0.0.2",
     "@hippy/hippy-live-reload-polyfill": "^1.0.4",
+    "@hippy/hippy-react-refresh-webpack-plugin": "^0.5.5",
     "babel-loader": "^8.1.0",
     "bezier-easing": "^2.1.0",
     "case-sensitive-paths-webpack-plugin": "^2.2.0",
+    "cross-env": "^7.0.3",
```
Comment (zoomchan-cxj) [inline examples/hippy-react-demo/package.json:55]: 移除

File: examples/hippy-react-demo/scripts/hippy-webpack.dev.js [hunk 5]
```diff
@@ -2,14 +2,27 @@ const fs = require('fs');
...
+    hot: true,
+    devMiddleware: {
+      writeToDisk: true,
+    },
+    client: {
+      overlay: false,
+    },
+  },
   entry: {
     index: ['regenerator-runtime', path.resolve(pkg.main), '@hippy/hippy-live-reload-polyfill'],
```
Comment (zoomchan-cxj) [inline examples/hippy-react-demo/scripts/hippy-webpack.dev.js:27]: '@hippy/hippy-live-reload-polyfill'  可以删除

File: examples/hippy-react-demo/package.json [hunk 6]
```diff
@@ -8,23 +8,23 @@
   "description": "Demo app for hippy-react",
   "repository": "https://github.com/Tencent/Hippy/tree/master/examples/hippy-react-demo",
   "scripts": {
-    "serve": "webpack-dev-server --mode=development --config ./scripts/hippy-webpack.web.js",
-    "build": "webpack --mode=production --config ./scripts/hippy-webpack.web.js",
```
Comment (zoomchan-cxj) [inline examples/hippy-react-demo/package.json:12]: web构建production保持跟原来一样，dev调试新建脚本，这样子不需要传 --mode=produciton

File: examples/hippy-react-demo/package.json [hunk 7]
```diff
@@ -8,23 +8,23 @@
...
-    "hippy:local-debug": "node ../../packages/hippy-debug-server/index.js --live",
-    "hippy:debug": "hippy-debug --live",
-    "hippy:dev": "webpack --config ./scripts/hippy-webpack.dev.js",
-    "hippy:vendor": "webpack --config ./scripts/hippy-webpack.ios-vendor.js --config ./scripts/hippy-webpack.android-vendor.js",
-    "hippy:build": "webpack --config ./scripts/hippy-webpack.ios.js --config ./scripts/hippy-webpack.android.js"
+    "serve": "cross-env NODE_ENV=development webpack-dev-server --mode=development --config ./scripts/hippy-webpack.web.js",
```
Comment (zoomchan-cxj) [inline examples/hippy-react-demo/package.json:11]: serve 必然是调试的，也可以去掉 cross-env NODE_ENV=development 和 --mode

File: examples/hippy-react-demo/package.json [hunk 8]
```diff
@@ -8,23 +8,23 @@
...
+    "build": "cross-env NODE_ENV=production webpack --mode=production --config ./scripts/hippy-webpack.web.js",
+    "hippy:local-debug": "cross-env NODE_ENV=development node ../../packages/hippy-debug-server/index.js -c ./scripts/hippy-webpack.dev.js",
+    "hippy:debug": "cross-env NODE_ENV=development hippy-debug -c ./scripts/hippy-webpack.dev.js",
```
Comment (zoomchan-cxj) [inline examples/hippy-react-demo/package.json:14]: hippy-vue 加了--host, hippy-react 没加

File: examples/hippy-vue-demo/package.json [hunk 9]
```diff
@@ -12,8 +12,8 @@
     "serve": "vue-cli-service serve",
     "build": "vue-cli-service build",
     "lint": "vue-cli-service lint",
-    "hippy:local-debug": "node ../../packages/hippy-debug-server/index.js --live",
-    "hippy:debug": "hippy-debug --live",
+    "hippy:local-debug": "node ../../packages/hippy-debug-server/index.js -c scripts/hippy-webpack.dev.js --host 0.0.0.0",
+    "hippy:debug": "hippy-debug -c scripts/hippy-webpack.dev.js",
```
Comment (zoomchan-cxj) [inline examples/hippy-vue-demo/package.json:16]: debug不加 --host?

File: examples/hippy-react-demo/scripts/hippy-webpack.dev.js [hunk 10]
```diff
@@ -45,6 +65,7 @@ module.exports = {
           {
             loader: 'babel-loader',
             options: {
+              compact: false,
```
Comment (zoomchan-cxj) [inline examples/hippy-react-demo/scripts/hippy-webpack.dev.js:68]: react 加了compact： false了，vue是否也加？

File: packages/hippy-debug-server/package.json [hunk 11]
```diff
@@ -19,13 +22,93 @@
...
+    "execa": "^5.1.1",
+    "file-loader": "^6.2.0",
+    "html-webpack-plugin": "^4.5.2",
+    "http-proxy": "^1.18.1",
+    "husky": "^7.0.0",
+    "jest": "^27.0.4",
+    "klona": "^2.0.4",
+    "less": "^4.1.1",
+    "less-loader": "^7.3.0",
+    "lint-staged": "^12.0.2",
```
Comment (zoomchan-cxj) [inline packages/hippy-debug-server/package.json:86]: 无关的npm包去掉，如lint-stage、eslint 等等

File: packages/hippy-vue-loader/CHANGELOG.md [hunk 12]
```diff
@@ -0,0 +1,5635 @@
+<a name="2.11.5-alpha.4"></a>
```
Comment (zoomchan-cxj) [inline packages/hippy-vue-loader/CHANGELOG.md]: changelog去掉

File: packages/hippy-vue-loader/tsconfig.json [hunk 13]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (zoomchan-cxj) [inline packages/hippy-vue-loader/tsconfig.json:1]: tsconfig 是是否要保留

File: packages/hippy-vue/src/renderer/native/style/css-selectors.js [hunk 14]
```diff
@@ -127,6 +135,10 @@ class IdSelector extends SimpleSelector {
   lookupSort(sorter, base) {
     sorter.sortById(this.id, base || this);
   }
+
+  removeSort(sorter, base) {
```
Comment (zoomchan-cxj) [inline packages/hippy-vue/src/renderer/native/style/css-selectors.js:139]: removeSort(sorter, base)  写了两次

**Comment by zoomchan-cxj** [comment]
Vue-Loader 和 Webpack-dev-server 相对源码的改动在Readme里说明一下

File: examples/hippy-react-demo/package.json (hunk 4 above)
Comment (shenchaoran) [reply examples/hippy-react-demo/package.json]: had removed

File: examples/hippy-react-demo/scripts/hippy-webpack.dev.js (hunk 5 above)
Comment (shenchaoran) [reply examples/hippy-react-demo/scripts/hippy-webpack.dev.js]: had removed

File: examples/hippy-react-demo/package.json (hunk 6 above)
Comment (shenchaoran) [reply examples/hippy-react-demo/package.json:12]: add `hippy-webpack.web.dev.js` and dev, prod will use different config

File: examples/hippy-react-demo/package.json (hunk 8 above)
Comment (shenchaoran) [reply examples/hippy-react-demo/package.json]: removed --host option

File: examples/hippy-react-demo/package.json (hunk 7 above)
Comment (shenchaoran) [reply examples/hippy-react-demo/package.json]: had removed

File: examples/hippy-vue-demo/package.json (hunk 9 above)
Comment (shenchaoran) [reply examples/hippy-vue-demo/package.json]: removed all --host option, use default 0.0.0.0

File: packages/hippy-debug-server/package.json (hunk 11 above)
Comment (shenchaoran) [reply packages/hippy-debug-server/package.json]: had removed

File: packages/hippy-vue-loader/CHANGELOG.md (hunk 12 above)
Comment (shenchaoran) [reply packages/hippy-vue-loader/CHANGELOG.md]: had removed

File: packages/hippy-vue-loader/tsconfig.json (hunk 13 above)
Comment (shenchaoran) [reply packages/hippy-vue-loader/tsconfig.json]: had removed

File: examples/hippy-react-demo/scripts/hippy-webpack.dev.js (hunk 10 above)
Comment (shenchaoran) [reply examples/hippy-react-demo/scripts/hippy-webpack.dev.js:71]: this is used for minify code of babel for debug, had removed this option

File: packages/hippy-debug-server/package.json [hunk 15]
```diff
@@ -19,13 +23,58 @@
...
+  },
+  "devDependencies": {
+    "@babel/cli": "^7.14.5",
+    "@babel/core": "^7.14.6",
+    "@babel/eslint-parser": "^7.14.7",
+    "@babel/plugin-transform-object-assign": "^7.14.5",
+    "@babel/plugin-transform-runtime": "^7.14.5",
+    "@babel/preset-env": "^7.14.5",
+    "@babel/runtime": "^7.14.5",
+    "@commitlint/cli": "^15.0.0",
```
Comment (zoomchan-cxj) [inline packages/hippy-debug-server/package.json:64]: 移除@commitlint 相关

File: examples/hippy-react-demo/scripts/hippy-webpack.web.dev.js [hunk 16]
```diff
@@ -0,0 +1,111 @@
+const fs = require('fs');
+const path = require('path');
+const webpack = require('webpack');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const CaseSensitivePathsPlugin = require('case-sensitive-paths-webpack-plugin');
+const ReactRefreshWebpackPlugin = require('@hippy/hippy-react-refresh-webpack-plugin');
```
Comment (zoomchan-cxj) [inline examples/hippy-react-demo/scripts/hippy-webpack.web.dev.js:6]: web应该不需要hippy的插件？

Comment (shenchaoran) [reply examples/hippy-react-demo/scripts/hippy-webpack.web.dev.js:6]: web 需要用 `@pmmmwh/react-refresh-webpack-plugin` ，引用 `@hippy/hippy-react-refresh-webpack-plugin` 也可以，后者加了个 test option，可以正则匹配 loader 作用的文件名

File: packages/hippy-debug-server/package.json (hunk 15 above)
Comment (shenchaoran) [reply packages/hippy-debug-server/package.json]: removed

File: examples/hippy-vue-demo/package.json [hunk 17]
```diff
@@ -12,9 +12,10 @@
     "serve": "vue-cli-service serve",
     "build": "vue-cli-service build",
     "lint": "vue-cli-service lint",
-    "hippy:local-debug": "node ../../packages/hippy-debug-server/index.js --live",
-    "hippy:debug": "hippy-debug --live",
-    "hippy:dev": "webpack --config ./scripts/hippy-webpack.dev.js",
+    "hippy:local-debug": "node ../../packages/hippy-debug-server/index.js",
+    "hippy:local-dev": "node ../../packages/hippy-debug-server/hippy-dev.js -c ./scripts/hippy-webpack.dev.js",
+    "hippy:debug": "hippy-debug",
+    "hippy:dev": "hippy-dev -c ./scripts/hippy-webpack.dev.js",
```
Comment (zoomchan-cxj) [inline examples/hippy-vue-demo/package.json:18]: -c 文档里说明一下

File: packages/hippy-debug-server/README.md [hunk 18]
```diff
@@ -4,6 +4,11 @@
 
 The package provide the debug server communicated with native(Android apk or iOS Simulator) and the local web development.
 
+# Introduction
+Some of source code is fork from other project. we modify the client implemention so the injected js could run on Hippy.
```
Comment (zoomchan-cxj) [inline packages/hippy-debug-server/README.md:8]: forked 过去式

File: packages/hippy-debug-server/README.md [hunk 19]
```diff
@@ -4,6 +4,11 @@
...
+- The webpack-dev-server folder if fork from [webpack-dev-server](https://github.com/webpack/webpack-dev-server). We modify all platform related implemention to support Hippy, such like `window`, `localtion`, etc.
```
Comment (zoomchan-cxj) [inline packages/hippy-debug-server/README.md:9]: if fork => is forked

File: packages/hippy-vue-loader/README.md [hunk 20]
```diff
@@ -0,0 +1,49 @@
+# Hippy Vue Loader
+
+This is a fork of [vue-loader@15.9.8](https://github.com/vuejs/vue-loader/tree/master) for support hippy css HMR.
```
Comment (zoomchan-cxj) [inline packages/hippy-vue-loader/README.md:3]: support => supporting

File: packages/hippy-vue-loader/README.md [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
+
+Add [style hot reload hook](./lib/codegen/hotReload.js) and [repaint API](./lib/vue-hot-reload-api/index.js) to repaint component when HMR.
+
+
+
+# How to use
+This loader should use together with [hippy hmr plugin](https://github.com/hippy-contrib/hippy-hmr-plugin) and [hipp debug server](https://github.com/Tencent/Hippy/tree/master/packages/hippy-debug-server), an example webpack config like this:
```
Comment (zoomchan-cxj) [inline packages/hippy-vue-loader/README.md:10]: should use =>  should be used

File: packages/hippy-debug-server/README.md [hunk 22]
```diff
@@ -4,6 +4,11 @@
 
 The package provide the debug server communicated with native(Android apk or iOS Simulator) and the local web development.
 
+# Introduction
+Some of source code is fork from other project. we modify the client implemention so the injected js could run on Hippy.
+- The webpack-dev-server folder if fork from [webpack-dev-server](https://github.com/webpack/webpack-dev-server). We modify all platform related implemention to support Hippy, such like `window`, `localtion`, etc. 
+- webpack-dev-server/client-src/hot is fork from [webpack](https://github.com/webpack/webpack/tree/main/hot). We change the fallback strategy to [live-reload](./webpack-dev-server/hot/../client-src/utils/apply-reload.js) of Hippy.
```
Comment (zoomchan-cxj) [inline packages/hippy-debug-server/README.md:10]: is forked

**Review by zoomchan-cxj — APPROVED** [review]
LGTM

**Pull request merged**